src/D2L.CodeStyle.UnsafeStaticCounter/Program.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace D2L.CodeStyle.UnsafeStaticCounter {
+	internal sealed class Program {
+		const int DEFAULT_MAX_CONCURRENCY = 4;
+
+		static void Main( string[] args ) {
+			Task.Run( async () => await AsyncMain( args ) )
+				.GetAwaiter()
+				.GetResult();
+		}
+
+		static async Task AsyncMain( string[] args ) {
+			try {
+				await AsyncThrowableMain( args );
+			} catch( Exception e ) {
+				LogException( e );
+			}
+		}
+
+		static async Task AsyncThrowableMain( string[] args ) {
+			var options = ParseOptions( args );
+			var prog = new Counter( options );
+			await prog.Run();
+		}
+
+		static void LogException( Exception e ) {
+			while( e != null ) {
+				if( e is AggregateException ) {
+					var ae = e as AggregateException;
+					foreach( var ie in ae.InnerExceptions ) {
+						LogException( ie );
+					}
+				} else {
+					Console.WriteLine( $"error: {e.Message}" );
+				}
+				e = e.InnerException;
+			}
+		}
+
+		private static Options ParseOptions( string[] args ) {
+			string path = @"c:\d2l\instances\all\checkout";
+			int concurrency = DEFAULT_MAX_CONCURRENCY;
+			string outputFile = "statics.json";
+
+			var enumerator = ( (IEnumerable<string>)args ).GetEnumerator(); // Array has non-generic enumerator only :facepalm:
+			while( enumerator.MoveNext() ) {
+				switch( enumerator.Current ) {
+					case "-n":
+						enumerator.MoveNext();
+						concurrency = int.Parse( enumerator.Current );
+						break;
+					case "-d":
+						enumerator.MoveNext();
+						path = enumerator.Current;
+						break;
+					case "-o":
+						enumerator.MoveNext();
+						outputFile = enumerator.Current;
+						break;
+				}

```

Comment:
Have you ever seen this guy?  

https://git.dev.d2l/projects/TOOL/repos/ndeskoptionstypedoptionset/browse/NDesk.Options.TypedOptionSet.UnitTests/OptionSetTests.cs